### ISSUE TITLE
Passing in RunSettings on the CLI crashes

### ISSUE THREAD

**Issue by ViktorHofer** [microsoft/vstest#2360]
## Description
If you pass in RunSettings via the CLI, the parser crashes.

## Steps to reproduce
> What steps can reproduce the defect?
> Please share the setup, commandline for vstest.console, sample project, target
> framework etc.

```
C:\git\runtime3\artifacts\bin\System.Text.RegularExpressions.Tests\netcoreapp5.0-Debug>  "C:\git\runtime3\.dotnet\dotnet" test System.Text.RegularExpressions.Tests.dll --logger "xunit;LogFilePath=testResults.xml" --platform x64 --parallel --filter "(category!=OuterLoop&category!=failing&category!=nonnetcoreapptests&category!=nonwindowstests)"  --  RunConfiguration.DisableAppDomain=true
```

## Expected behavior
> Share the expected output

`Starting test execution, please wait...`

## Actual behavior
> What is the behavior observed?

Test invocation runs successfully.

## Diagnostic logs
> Please share test platform diagnostics logs. Instructions to collect logs are [here](https://github.com/Microsoft/vstest-docs/blob/master/docs/diagnose.md#test-platform-diagnostics).  
> The logs may contain test assembly paths, kindly review and mask those before sharing.

## Environment
> Please share additional details about the test environment.
> Operating system, Build version of vstest.console

Microsoft (R) Test Execution Command Line Tool Version 16.6.0-preview-20200310-03
SDK: `5.0.100-preview.2.20163.4`

cc @nohwnd

**Comment by nohwnd** [comment]
This was partially fixed [here](  https://github.com/dotnet/sdk/pull/10883 ), to enable @ViktorHofer scenario. But would still benefit from refactoring tha parser in sdk to: 
- don't search all parameters for dll / exe (which enables the new dotnet test abc.dll functionality), but instead search only the identified sources
- use the part after `--` literally, and don't interpret anything in it
- consider using actual parsing library like the CommandLine library 

@pavelhorak  Pushing this to 16.7.0, it was added to 16.6.0 milestone because it was blocker to another scenario and it is not anymore so we can tackle this in the next one.

**Comment by nohwnd** [comment]
Will address this after #862 is fixed because there is a bit of overlap.

**Comment by nohwnd** [comment]
Investigated this and the benefits are very small, while rewriting the code is a quite big change. Closing as the most important part was addressed elsewhere and the above command works in `5.0.100-preview.5.20255.2` sdk and newer.

- don't search all parameters for dll / exe (which enables the new dotnet test abc.dll functionality), but instead search only the identified sources <- we don't take dll in any other param, the benefits of this are small
- use the part after `--` literally, and don't interpret anything in it <- this was addressed in #2409 
- consider using actual parsing library like the CommandLine library <- this would be nice, but I don't see any benefit right now apart from being "cleaner"
